conf/nginx/nginx.conf

```diff
@@ -0,0 +1,32 @@
+error_log stderr debug;
+
+events {
+    worker_connections 1024;
+}
+
+http {
+    server {
+        listen _XJENKINS_PORT default_server;
+
+        access_log      /var/log/nginx/jenkins/access.log;
+        error_log       /var/log/nginx/jenkins/error.log;
+
+        location ^~ _XJENKINS_CONTEXT {
+            proxy_pass         http://127.0.0.1:8080;
+            proxy_set_header   Host             $http_host;
+            proxy_set_header   X-Real-IP        $remote_addr;
+            proxy_set_header   X-Forwarded-For  $proxy_add_x_forwarded_for;
+            proxy_max_temp_file_size 0;
+        }
+
+        location ~ ^/(?<url>.*)$ {
+            rewrite ^/(?<url>.*)$ _XJENKINS_CONTEXT/$url break;
+            proxy_pass         http://127.0.0.1:8080;
```

Comment:
Jenkins is running inside the container but we deploy the container with host networking so that the Mesos master can communicate with the scheduler. Therefore, we probably shouldn't hard-code this to `127.0.0.1:8080` since there's a very good chance some other service will try using this port on the agent.